gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java

```diff
@@ -138,8 +138,13 @@ protected List<FileStatus> getFilesAtPath(FileSystem fs, Path path, PathFilter f
   private List<FileStatus> recursivelyGetFilesAtDatePath(FileSystem fs, Path path, String traversedDatePath, PathFilter fileFilter,
       int level,  LocalDateTime startDate, LocalDateTime endDate, DateTimeFormatter formatter) throws IOException {
     List<FileStatus> fileStatuses = Lists.newArrayList();
-    Iterator<FileStatus> folderIterator = Arrays.asList(fs.listStatus(path)).iterator();
-
+    Iterator<FileStatus> folderIterator;
+    try {
+      folderIterator = Arrays.asList(fs.listStatus(path)).iterator();
```

Comment:
why not add check before list status but rely on the exception? 
what happens if it's some error on the fs side? isn't it better for us to throw exception to fail the job instead of returning empty list?

Reply:
By check before list status do you mean to check that all the folders match the date time format at that current level of granularity? Just following the previous behavior of this finder, which was to report an exception and return empty list instead of failing the job. Current issues are caused by failling the job and not ignoring badly formatted files.